sdk/python/lib/pulumi/provider/server.py

```diff
@@ -126,6 +127,8 @@ async def Construct(
         try:
             return await self._construct(request, context)
         except Exception as e:  # noqa
+            if isinstance(e, RunError):
+                raise
```

Comment:
`RunError` typically means we want a clean exit, and we don't want to print the full stack trace.

I noticed that while working on this issue that when using a python program we'd print a stack trace within our construct code during shutdown, but we don't in Node.js.

Reply:
Is it possible to write a test for this?

Reply:
I added a set of tests for the various Construct error types https://github.com/pulumi/pulumi/pull/19976/commits/909b7e911b24d0f982e7e35f52ce1492affc9311